src/Okta.Sdk/Internal/UserAgentBuilder.cs

```diff
@@ -22,12 +23,15 @@ public sealed class UserAgentBuilder
 
         // Lazy ensures this only runs once and is cached.
         private readonly Lazy<string> _cachedUserAgent;
+        private readonly ILogger _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="UserAgentBuilder"/> class.
         /// </summary>
-        public UserAgentBuilder()
+        /// <param name="logger">An optional logging interface</param>
+        public UserAgentBuilder(ILogger logger = null)
```

Comment:
The typical pattern is for classes to accept an `ILogger` whose type parameter is the class itself. That way the logging framework can automatically trace the name of the class along with any log messages.
```cs
public UserAgentBuilder(ILogger<UserAgentBuilder> logger)
{
  ...
}
```

If you are using the `Microsoft.Extensions.DependencyInjection` framework the correct kind of logger can be automatically created and passed to the constructor. If you aren't, you can use an [ILoggerFactory](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.iloggerfactory?view=aspnetcore-2.1) to create the right kind of logger by calling [CreateLogger&lt;T&gt;()](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerfactoryextensions.createlogger?view=aspnetcore-2.1#Microsoft_Extensions_Logging_LoggerFactoryExtensions_CreateLogger__1_Microsoft_Extensions_Logging_ILoggerFactory_).

Reply:
We could flip to `ILogger<T>`, but `NullLogger<T>` isn't available until (I think) 2.1?  So I skipped this for now.

Reply:
Yeah I see other places in the code where `ILogger` is being used instead of `ILogger<T>`, so this is probably something the devex team should consider changing. I guess that goes a bit beyond the scope of your change.

I'm not on the devex team, BTW, just helping with .NET reviews. 😄